### PR TITLE
Ili9341 doc update

### PR DESCRIPTION
**Pull request by kbx81**
## Description:

Updates #723 per changes requested there by @glmnet 

## Checklist:

  - [x] Branch: `next` is for changes and new documentation that will go public with the next ESPHome release. Fixes, changes and adjustments for the current release should be created against `current`.
  - [x] Link added in `/index.rst` when creating new documents for new components or cookbook.

### CODE REVIEW

**Comment by kbx81** [comment]
FWIW, yesterday I was able to test this component and documentation on the ESP-WROVER-KIT that I have and it appears to work as expected.￼ 🎉

File: components/display/ili9341.rst [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+      - platform: ili9341
+        model: TFT 2.4
+        cs_pin: 14
+        dc_pin: 27
+        led_pin: 32  ### see note below ###
+        reset_pin: 33
+        rotation: 0
+        
+        lambda: |-
+          it.fill(ili9341::COLOR_BLACK);
```
Comment (glmnet) [inline components/display/ili9341.rst]: Does this works? Should it work?

File: components/display/ili9341.rst [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+          it.print(0, 0, id(font), "Hello World!");
```
Comment (glmnet) [inline components/display/ili9341.rst]: It'd be nice to showcase color print?

File: components/display/ili9341.rst [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+
+    # Example minimal configuration entry
+    display:
+      - platform: ili9341
+        model: TFT 2.4
+        cs_pin: 14
+        dc_pin: 27
+        led_pin: 32  ### see note below ###
+        reset_pin: 33
+        rotation: 0
```
Comment (glmnet) [inline components/display/ili9341.rst]: If rotation 0 is default remove it from here. On the first example we tend to do the minimum working configuration for a component, so usually optional parameters are not specified.

**Review by glmnet — COMMENTED** [review]
Looks good! Thanks for reviewing specially the samples given works. People always expect to copy paste code and get it working right away

File: components/display/ili9341.rst (hunk 1 above)
Comment (kbx81) [reply components/display/ili9341.rst]: I will confirm...could be that I missed part of it. I'll check and fix as necessary.

File: components/display/ili9341.rst (hunk 2 above)
Comment (kbx81) [reply components/display/ili9341.rst]: Agreed -- will update.

File: components/display/ili9341.rst (hunk 3 above)
Comment (kbx81) [reply components/display/ili9341.rst]: Consider it done. 😄

File: components/display/ili9341.rst (hunk 1 above)
Comment (kbx81) [reply components/display/ili9341.rst]: A find/replace got the better of me. This is fixed and confirmed to work.

**Comment by glmnet** [comment]
Thanks!

**Pull request merged**